pkg/pillar/types/certinfotypes.go

```diff
@@ -30,24 +30,24 @@ type ControllerCertStatus struct {
 	Type     zcert.ZCertType
 	Cert     []byte
 	CertHash []byte
-	ErrorInfo
+
+	Error     string
+	ErrorTime time.Time
```

Comment:
Why are you going in the reverse? Is it not better for all the structures to use ErrInfo to carry the Error information?
Then the Set / Clear functions as well as any encode functions can be shared. 

What is the need for having these fields separate?

Reply:
@kalyan-nidumolu the purpose is to avoid further confusion and misuse, by being very explicit about what is needed. 

Having an ErrorWithTIme structure might make sense, but applying that across the board would be a fair bit of change.
But I'll take a look at it.

Reply:
Done